cfme/services/requests.py

```diff
@@ -1,286 +1,370 @@
+        return self.in_requests
+
+
+class RequestDetailsToolBar(RequestView):
+    copy = Button(title='Copy original Request')
+    edit = Button(title='Edit the original Request')
+    delete = Button(title='Delete this Request')
+    reload = Button(title='Reload the current display')
+    approve = Button(title='Approve this Request')
+    deny = Button(title='Deny this Request')
+
+
+class RequestDetailsView(RequestView):
+    @View.nested
+    class details(View):  # noqa
+        request_id = SummaryFormItem('Request Details', 'Request ID')
+        status = SummaryFormItem('Request Details', 'Status')
+        request_state = SummaryFormItem('Request Details', 'Request State')
+        requester = SummaryFormItem('Request Details', 'Requester')
+        request_type = SummaryFormItem('Request Details', 'Request Type')
+        description = SummaryFormItem('Request Details', 'Description')
+        last_message = SummaryFormItem('Request Details', 'Last Message')
+        created_on = SummaryFormItem('Request Details', 'Created On')
+        last_update = SummaryFormItem('Request Details', 'Last Update')
+        completed = SummaryFormItem('Request Details', 'Completed')
+        approval_state = SummaryFormItem('Request Details', 'Approval State')
+        approved_by = SummaryFormItem('Request Details', 'Approved/Denied by')
+        approved_on = SummaryFormItem('Request Details', 'Approved/Denied on')
+        reason = SummaryFormItem('Request Details', 'Reason')
+        provisioned_vms = SummaryFormItem('Request Details', 'Provisioned VMs')
+
+    @View.nested
+    class request(Tab):  # noqa
+        TAB_NAME = 'Request'
+        email = SummaryFormItem('Request Information', 'E-Mail')
+        first_name = SummaryFormItem('Request Information', 'First Name')
+        last_name = SummaryFormItem('Request Information', 'Last Name')
+        notes = SummaryFormItem('Request Information', 'Notes')
+        manager_name = SummaryFormItem('Manager', 'Name')
+
+    @View.nested
+    class purpose(Tab):  # noqa
+        TAB_NAME = 'Purpose'
+        apply_tags = BootstrapTreeview('all_tags_treebox')
+
+    @View.nested
+    class catalog(Tab):  # noqa
+        TAB_NAME = 'Catalog'
+        filter = SummaryFormItem('Select', 'Filter')
+        name = SummaryFormItem('Select', 'Name')
+        provision_type = SummaryFormItem('Select', 'Provision Type')
+        linked_clone = Checkbox(id='service__linked_clone')
+        vm_count = SummaryFormItem('Number of VMs', 'Count')
+        instance_count = SummaryFormItem('Number of Instances', 'Count')
+        vm_name = SummaryFormItem('Naming', 'VM Name')
+        instance_name = SummaryFormItem('Naming', 'Instance Name')
+        vm_description = SummaryFormItem('Naming', 'VM Description')
+
+    @View.nested
+    class environment(Tab):  # noqa
+        TAB_NAME = 'Environment'
+
+        automatic_placement = Checkbox(name='environment__placement_auto')
+        # Azure
```

Comment:
not needed here. on this view elements are read, but it's a cool hint to update some other views. thanks